docs/hub/datasets-dask.md

```diff
@@ -45,3 +61,56 @@ df_test  = dd.read_parquet("hf://datasets/username/my_dataset/test")
 ```
 
 For more information on the Hugging Face paths and how they are implemented, please refer to the [the client library's documentation on the HfFileSystem](/docs/huggingface_hub/guides/hf_file_system).
+
+# Process data
+
+To process a dataset in parallel using Dask, you can first define your data processing function for a pandas DataFrame or Series, and then use the Dask `map_partitions` function to apply this function to all the partitions of a dataset in parallel:
+
+```python
+def dummy_count_words(texts):
+    return pd.Series([len(text.split(" ")) for text in texts])
+```
+
+In pandas you can use this function on a text column:
+
+```python
+# pandas API
+df["num_words"] = dummy_count_words(df.text)
+```
+
+And in Dask you can run this function on every partition:
+
+```python
+# Dask API: run the function on every partition
+df["num_words"] = df.text.map_partitions(dummy_count_words, meta=int)
+```
+
+Note that you also need to provide `meta` which is the type of the pandas Series or DataFrame in the output of your function.
+This is needed because Dask DataFrame is a lazy API. Since Dask will only run the data processing once `.compute()` is called, it needs
+the `meta` argument to know the type of the new column in the meantime.
+
+# Predicate and Projection Pushdown
+
+When reading Parquet data from Hugging Face, Dask automatically leverages the metadata in Parquet files to skip entire files or row groups if they are not needed. For example if you apply a filter (predicate) on a Hugging Face Dataset in Parquet format or if you select a subset of the columns (projection), Dask will read the metadata of the Paquet files to discard the parts that are not needed without downloading them.
+
+This is possible thanks to the `dask-expr` package which is generally installed by default with Dask.
+
+For example this subset of FineWeb-Edu contains many Parquet files. If you can filter the dataset to keep the text from recent CC dumps, Dask will skip most of the files and only download the data that match the filter:
+
+```python
+import dask.dataframe as dd
+
+df = dd.read_parquet("hf://datasets/HuggingFaceFW/fineweb-edu/sample/10BT/*.parquet")
+
+# Dask will skip the files or row groups that don't
+# match rhe query without downloading them.
+df = df[df.dump >= "CC-MAIN-2023"]
```

Comment:
Does Dask not still need to download the data to check the values in this column match this filter? From what I understood in the Polars case the predicate push down is usually used for skipping the reading of a column i.e. if you drop it later it doesn't bother to load it and/or doing a filtering step early on. Is Dask directly able to do this before loading?

Reply:
so it will skip the row groups which don't have any row that match the query using the row group metadata

then on the remaining row groups it will download the column used for filtering to apply the filter

the other columns will be downloaded or not based on the other operations done on the dataset

Reply:
> if you drop it later it doesn't bother to load it and/or doing a filtering step early on. Is Dask directly able to do this before loading?

yes correct ! would be cool to explain that here as well